src/main/java/build/buildfarm/common/DigestUtil.java

```diff
@@ -18,24 +18,43 @@
 import build.bazel.remote.execution.v2.Digest;
 import build.bazel.remote.execution.v2.DigestFunction;
 import build.buildfarm.common.blake3.Blake3HashFunction;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.hash.Funnels;
 import com.google.common.hash.HashCode;
 import com.google.common.hash.Hasher;
 import com.google.common.hash.Hashing;
 import com.google.common.hash.HashingOutputStream;
 import com.google.common.io.ByteSource;
 import com.google.protobuf.ByteString;
-import com.google.protobuf.Descriptors.EnumValueDescriptor;
 import com.google.protobuf.Message;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Set;
 import lombok.Getter;
 
 /** Utility methods to work with {@link Digest}. */
 public class DigestUtil {
```

Comment:
This class could probably benefit from more unit tests to prevent any regressions